### PR TITLE
fix: remove sensitive information from the logs - WPB-1705

### PR DESCRIPTION
**Pull request by johnranj**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-1705" title="WPB-1705" target="_blank">BugWPB-1705</a>  [wire-ios-mono] Prevent logging of sensitve data in "developer" or "internal" build
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

Remove sensitive information like token informations on the logs

### Solutions

Removed logs which contain sensitive information

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Review by MarcinR — APPROVED** [review]

File: wire-ios-transport/Source/Authentication/ZMAccessTokenHandler.m [hunk 1]
```diff
@@ -304,7 +300,6 @@ - (BOOL)processAccessTokenResponse:(ZMTransportResponse *)response
         
         if ( ! didFail) {
             self.accessToken = newToken;
-            ZMLogInfo(@"New access token <%@: %p>", self.accessToken.class, self.accessToken);
```
Comment (netbe) [inline wire-ios-transport/Source/Authentication/ZMAccessTokenHandler.m:307]: should we keep this info for debug build (around #if DEBUG macro with WireLogger) ?

File: wire-ios-transport/Source/TransportSession/ZMTransportSession.m [hunk 2]
```diff
@@ -503,9 +503,6 @@ - (void)didCompleteRequest:(ZMTransportRequest *)request data:(NSData *)data tas
     [self.remoteMonitoring logWithResponse:httpResponse];
 
     ZMLogDebug(@"ConnectionProxyDictionary: %@,", session.configuration.connectionProxyDictionary);
-    ZMLogPublic(@"Response to %@: %@", request.safeForLoggingDescription,  response.safeForLoggingDescription);
-    ZMLogInfo(@"<---- Response to %@ %@ (status %u): %@", [ZMTransportRequest stringForMethod:request.method], request.path, (unsigned) httpResponse.statusCode, response);
-    ZMLogInfo(@"URL Session is %@", session.description);
```
Comment (netbe) [inline wire-ios-transport/Source/TransportSession/ZMTransportSession.m:508]: here I think we have the safe correct info in DataDog. so it's ok to just delete 👍

File: wire-ios-transport/Source/Authentication/ZMAccessTokenHandler.m [hunk 3]
```diff
@@ -191,7 +191,6 @@ - (void)sendAccessTokenRequestWithURLSession:(ZMURLSession *)URLSession clientID
         return;
     }
 
-    ZMLogInfo(@"Requesting access token from cookie (existing token: %p).", self.accessToken);
```
Comment (netbe) [inline wire-ios-transport/Source/Authentication/ZMAccessTokenHandler.m:194]: I would put it as available for debug build with WireLogger

Comment (johnranj) [reply wire-ios-transport/Source/Authentication/ZMAccessTokenHandler.m:194]: I assume, even WireLogger logs will be available on the Internal builds

Comment (netbe) [reply wire-ios-transport/Source/Authentication/ZMAccessTokenHandler.m:194]: true, I haven't looked at the implementation, but end result should be log visible in console with Debug configuration (not on DataDog)

**Review by netbe — APPROVED** [review]

**Pull request merged**